src/components/content/MessageWithPopper.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+    Box,
+    Button,
+    ClickAwayListener,
+    Fade,
+    Popper,
+    useMediaQuery,
+    useTheme,
+} from '@mui/material';
+import { logDialogBackground, popperIndex } from 'context/Theme';
+import { MouseEvent, ReactNode, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+interface Props {
+    messageId: string;
+    popper: ReactNode;
+}
+
+function MessageWithPopper({ messageId, popper }: Props) {
+    const theme = useTheme();
+    const belowMd = useMediaQuery(theme.breakpoints.down('md'));
+
+    const [open, setOpen] = useState<boolean>(false);
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const canBeOpen = open && Boolean(anchorEl);
+    const id = canBeOpen ? 'transition-popper' : undefined;
+
+    const handlers = {
+        togglePopper: (event: MouseEvent<HTMLElement>) => {
+            setAnchorEl(event.currentTarget);
+            setOpen((previousOpen) => !previousOpen);
+        },
+        externalAreaClicked: () => {
+            setOpen(false);
+        },
+    };
+
+    return (
+        <FormattedMessage
+            id={messageId}
+            tagName={Box}
+            values={{
+                buttonLabel: (
+                    <>
+                        <Button
+                            variant="text"
+                            size="small"
+                            onClick={handlers.togglePopper}
+                            sx={{ mb: 0.5 }}
+                        >
+                            <FormattedMessage id={`${messageId}.buttonLabel`} />
+                        </Button>
+
+                        <Popper
```

Comment:
`MesageWithPopper` was phased out of integration by `ButtonWithPopper`, but the code feels worthwhile enough to leave behind. Both components involve a popper with the same component hierarchy. Given `MessageWithPopper` is unused at the moment, I debated whether it would be worthwhile to create a popper wrapper component. Thoughts?

Reply:
We probably should make a wrapper.